### ISSUE TITLE
rmulti() incorrect correlation matrix subsetting 

### ISSUE THREAD

**Issue by yann1cks** [debruine/faux#107]
Hello,

thank you very much for the great package! rmulti() is a real time-saver for my thesis. 

I think I encountered a bug for simulations with more than 3 parameters. In this case the correlation matrix is subsetted incorrectly. Here is an example of using rmulti() with 5 parameters. As you can see the returned correlation matrix of the simulated data and the correlation matrix passed to the function are not the same. In the example I am using `empirical = TRUE`, because the error in the correlation matrix is easier to spot.

```
# Import and set seed
library(faux)
set.seed(123)

# Set up correlation matrix
r <- c(1, 0.1, 0.2, 0.3, 0.4,
       0.1, 1, 0.5, 0.6, 0.7,
       0.2, 0.5, 1, 0.8, 0.1,
       0.3, 0.6, 0.8, 1, 0.2,
       0.4, 0.7, 0.1, 0.2, 1)

# Simulate data. 
data <- faux::rmulti(
  n = 1000,
  dist = c(A = "norm", B = "norm", C = "norm", D = "norm",  E = "norm"),
  r = r,
  empirical = TRUE
)
cor(data)
#       A     B     C     D     E
# A 1.000 0.102 0.201 0.500 0.301
# B 0.102 1.000 0.602 0.799 0.400
# C 0.201 0.602 1.000 0.701 0.102
# D 0.500 0.799 0.701 1.000 0.201
# E 0.301 0.400 0.102 0.201 1.000
```

Looking at the rmulti() function I think the error in in line 144 `pairs$r <- cormat[upper.tri(cormat)]`. 
```
cormat = matrix(r, ncol=5)

cormat[upper.tri(cormat)]
#  0.1 0.2 0.5 0.3 0.6 0.8 0.4 0.7 0.1 0.2

cormat[lower.tri(cormat)]
#  0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.1 0.2 
```
As you can see subsetting the matrix with `upper.tri` returns an incorrect matrix. I am not 100%  why this is happening, but using `lower.tri` returns the correct result. I did not create a PR as I do not fully understand why this fixes the issue and I am not sure how tests in R work. But I can give it a try if you want.

Kind regards
Yannick

**Comment by debruine** [comment]
Thanks so much! I'll have a look at this as soon as I can and let you know what I figure out.

**Comment by DanSimonet** [comment]
Wanted to echo the comment love `faux` and believe this is the closest simulation package I have found which emulates all use cases (also dabbling with `fabricatr` and `simstudy`). 
 
I was going to report the same thing as yann1cks so will provide reproducible example. Basically `rmulti` and `rnorm_multi` return different results with the same input. 

```
cmat <- matrix(c(1, 0.30, 0.10,
                 0.30, 1, 0.30,
                 0.10, 0.30, 1), 3, 3)

set.seed(1)

perf_dat_1 <- rmulti(
  n = 20,
  dist = c(education = "norm",
           OA = "norm",
           job_perf = "norm"),
  params = list(education = c(mean = 15, sd = 3),
                OA = c(mean = 30, sd = 5),
                job_perf = c(mean = 4, sd = 1.5),
                r = cmat,
                empirical = TRUE
  ))

perf_dat_2 <- rnorm_multi(
  n = 20,
  mu = c(15, 30, 4),
  sd = c(3, 5, 1.5),
  r = cmat,
  varnames = c("education", "OA", "job_perf"),
  empirical = TRUE
)

get_params(perf_dat_1); get_params(perf_dat_2)
```
Running above returns the following with the first table offset but the second table being correct. Hoping to leverage `rmulti` to generate some matrices intermixing multiple variable types. 

| ID  | n | var | education | OA |job_perf | mean | sd |
|----|---|----|------------|----|---------|-------|---|
| 1 |20 |education |     1.00| -0.28 |    0.30| 15.42 |2.43|
| 2 |20 |       OA   |  -0.28 | 1.00  |  -0.22 |29.97| 4.36|
| 3 |20 | job_perf |     0.30 |-0.22|     1.00 | 4.29| 1.37|

|ID| n| var|    education|  OA| job_perf| mean|  sd|
|--|--|----|------------|-----|---------|------|----|
|1| 20| education |      1.0| 0.3|      0.1 |  15| 3.0|
|2 |20 |       OA     |  0.3 |1.0    |  0.3  | 30| 5.0|
|3 |20  |job_perf    |   0.1| 0.3   |   1.0|    4| 1.5|

**Comment by debruine** [comment]
@DanSimonet sorry this took so long to get to! Your code runs, but is malformed.

```
perf_dat_1 <- rmulti(
  n = 20,
  dist = c(education = "norm",
           OA = "norm",
           job_perf = "norm"),
  params = list(education = c(mean = 15, sd = 3),
                OA = c(mean = 30, sd = 5),
                job_perf = c(mean = 4, sd = 1.5),
                r = cmat,
                empirical = TRUE
  ))
```

You set `r` and `empirical` inside the `params` list() item, rather than as arguments to `rmulti()`, so they had their default values of 0 and FALSE. Try this code instead:

```
perf_dat_1 <- rmulti(
    n = 20,
    dist = c(a = "norm",
             b = "norm",
             c = "norm"),
    params = list(a = c(mean = 15, sd = 3),
                  b = c(mean = 30, sd = 5),
                  c = c(mean = 4, sd = 1.5)
    ),
    r = cmat,
    empirical = TRUE
)
```

**Comment by debruine** [comment]
Fixed in version 1.2.1.9002 (submitting to CRAN soon)
